### PR TITLE
pass `OptionDescr` by reference to save memory and more

### PR DESCRIPTION
**Pull request by donno2048**

### CODE REVIEW

**Comment by donno2048** [comment]
@dbabokin

**Review by Vsevolod-Livinskij — CHANGES_REQUESTED** [review]
Thank you for the pull request! It is great to see that someone is interested in the project!
There are two places that require minor changes before this can be merged.
Please let me know if you have any questions or need any help.

File: scripts/run_gen.py [hunk 1]
```diff
@@ -1633,25 +1627,20 @@ def gen_and_test(num, makefile, lock, end_time, task_queue, stat, targets, blame
...
-                test.add_fail_run(test_run)
-                continue
-
-            if not test_run.run():
-                test.add_fail_run(test_run)
-                continue
-
-            test.add_success_run(test_run)
+            # Run only specified targets
+            if t.specs.name in targets:
```
Comment (Vsevolod-Livinskij) [inline scripts/run_gen.py:1632]: I think it makes more sense to skip targets that are not specified upfront.
In this case, we don't have an extra offset, and it makes it more straightforward that there is no other code to handle them.

**Review by Vsevolod-Livinskij — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 2]
```diff
@@ -15,3 +15,8 @@ csmith_statistics_*.log
 
 !*/
 .idea
+*.log
+.vs/
+build/
+out/
```
[the remark this answers is not in the document]
Comment (donno2048) [reply .gitignore:21]: When using visual studio, it adds this directory to store Cmake data

File: scripts/run_gen.py (hunk 1 above)
Comment (donno2048) [reply scripts/run_gen.py]: Done

File: src/options.cpp [hunk 3]
```diff
@@ -269,15 +269,15 @@ bool OptionParser::optionStartsWith(char *option, const char *test) {
 // This function handles command-line options in form of "-short_arg <value>"
 // and performs action(<value>)
 bool OptionParser::parseShortArg(size_t argc, size_t &argv_iter, char **&argv,
-                                 OptionDescr option) {
-    std::string short_arg = option.getShortArg();
-    bool has_value = option.hasValue();
-    auto action = option.getAction();
+                                 OptionDescr *option) {
```
[the remark this answers is not in the document]
Comment (donno2048) [reply src/options.cpp]: Done

**Pull request merged**